resize.js

```diff
@@ -15,218 +18,216 @@ var S3 = new AWS.S3();
-  if ( matches[5] !== undefined ) {
-    resolutionX *= parseInt(matches[5]);
-    resolutionY *= parseInt(matches[5]);
-  }
-  fileType = matches[6].toLowerCase();
- 
-  if ( supportedFileType(fileType) === null ) {
-    log('error','Filetype ' + fileType + ' is not supported');
-    res.writeHead(415, 'Unsupported media type');
-    res.end();
-    return;
-  }
-
-  log('info','Requesting file ' + fileName + ' in ' + fileType + ' format in a ' + resolutionX + 'x' + resolutionY + 'px resolution');
-
-  Image.checkCacheOrCreate(fileName, fileType, resolutionX, resolutionY, res);
+function supportedFileType(fileType) {
+    switch (fileType) {
+    case 'jpg':
+    case 'jpeg':
+    case 'jfif':
+    case 'jpe':
+        return 'image/jpeg';
+    case 'png':
+        return 'image/png';
+    default:
+        return null;
+    }
 }
-Image.checkCacheOrCreate = function(fileName, fileType, resolutionX, resolutionY, res) {
-    // Check if it exists in the cache
-    selectImage.get([fileName, resolutionX, resolutionY, supportedFileType(fileType)],function(err,data) { 
-      if ( !err && data ) {
-        // It is in the cache, so redirect to there
-        log('info','cache hit for ' + fileName + '.' + fileType + '(' + resolutionX + 'x' + resolutionY + 'px)');
-        res.writeHead(302, {'Location': data.url, 'Cache-Control': 'public'});
+
+var Image = {};
+Image.get = function (req, res) {
+    var matches = req.url.match(/^\/(.*)_(\d+)_(\d+)(_(\d+)x)?\.(.*)/);
+    if (matches === null) {
+        // Invalid URL
+        log('error', '404 Error for ' + res.url);
+        res.writeHead(404, 'File not found');
         res.end();
         return;
-      } else {
+    }
+
+    var fileName = matches[1], resolutionX = parseInt(matches[2], 10), resolutionY = parseInt(matches[3], 10);
+    if (matches[5] !== undefined) {
+        resolutionX *= parseInt(matches[5], 10);
+        resolutionY *= parseInt(matches[5], 10);
+    }
+    var fileType = matches[6].toLowerCase();
+
+    if (supportedFileType(fileType) === null) {
+        log('error', 'Filetype ' + fileType + ' is not supported');
+        res.writeHead(415, 'Unsupported media type');
+        res.end();
+        return;
+    }
+
+    log('info', 'Requesting file ' + fileName + ' in ' + fileType + ' format in a ' + resolutionX + 'x' + resolutionY + 'px resolution');
+
+    Image.checkCacheOrCreate(fileName, fileType, resolutionX, resolutionY, res);
+};
+Image.checkCacheOrCreate = function (fileName, fileType, resolutionX, resolutionY, res) {
+    // Check if it exists in the cache
+    selectImage.get([fileName, resolutionX, resolutionY, supportedFileType(fileType)], function (err, data) {
+        if (!err && data) {
+            // It is in the cache, so redirect to there
+            log('info', 'cache hit for ' + fileName + '.' + fileType + '(' + resolutionX + 'x' + resolutionY + 'px)');
+            res.writeHead(302, {'Location': data.url, 'Cache-Control': 'public'});
```

Comment:
You can better use 303 or 307 here, 302 is an old code that wasn't used correctly (both by browsers and webservers) so 303 and 307 where created to replace it.